src/app/webdav/blob-service.service.ts

```diff
@@ -0,0 +1,131 @@
+import { Injectable } from '@angular/core';
+import { AppConfig } from '../app-config/app-config';
+import webdav from 'webdav';
+import { DomSanitizer, SafeUrl } from '@angular/platform-browser';
+import { SessionService } from 'app/session/session.service';
+
+@Injectable()
+export class BlobService {
+
+  // TODO Check connection/login success?
+  private client: any;
+  // private defaultImage: SafeUrl;
+  private fileList: string;
+  private currentlyGettingList: Promise<boolean>;
+
+  constructor(private domSanitizer: DomSanitizer,
+    private sessionService: SessionService) {
+      this.connect();
+  }
+
+  /**
+   * Reinitialize the client for the nextcloud server
+   * @param password login password
+   */
+  public connect(password: string = this.sessionService.getCurrentUser().blobPasswordDec) {
+    // clear the promise that retrieves the root dir
+    this.currentlyGettingList = null;
+    this.fileList = null;
+
+    if (this.sessionService.getCurrentUser() != null) {
```

Comment:
if currentUser is null, wouldn't that crash the system if it's trying to set the default value `connect(password: string = this.sessionService.getCurrentUser().blobPasswordDec)`? Maybe better to move that assignment into the method here inside the if block.